### PR TITLE
feat(kicad-mcp-pro): implement empty project quality gates and ipc lifecycle transition checks

### LINKED ISSUE #223: [BUG] IPC status reporting inconsistency — server_info says ipcAvailable:false but tools report 'IPC is reachable'
## Problem

There is a contradictory IPC status between `kicad_get_server_info` and the runtime tool error messages.

**`kicad_get_server_info` reports:**
```json
{
  "ipcAvailable": false,
  "ipcVersion": null,
  "livePcbContext": false,
  "liveSchematicContext": false
}
```

**But tool error messages say:**
> "KiCad IPC is **reachable**, but no PCB is open in the active KiCad session."

These are contradictory: `ipcAvailable: false` implies no IPC connectivity at all, but the tool diagnostics say IPC _is_ reachable.

Additionally, KiCad has both schematic and PCB editors open (confirmed by lock files `~test-kicad.kicad_pcb.lck` and `~test-kicad.kicad_sch.lck`), yet `livePcbContext` and `liveSchematicContext` are both `false`.

**Impact:** MCP clients that check `ipcAvailable` to decide their workflow will incorrectly assume IPC is fully down, when in fact it's partially available.

**Environment:**
- KiCad MCP Pro: v3.5.2
- KiCad CLI: 10.0.3
- OS: Windows 11
- Date: 27/05/2026

## Required changes

1. Align `ipcAvailable` in server-info with the actual IPC probe result at call time. If the IPC port responds, `ipcAvailable` should be `true`.
2. Introduce a separate field (e.g., `ipcDocumentLoaded: false`) to distinguish "IPC works but no document open" from "IPC port unreachable".
3. Ensure `livePcbContext` / `liveSchematicContext` update when documents are opened/closed.
4. If a lazy probe is used, document in server-info that values may be stale and provide a refresh mechanism.

## Validation

```bash
grep -rn "ipcAvailable\|livePcbContext" packages/mcp-server/src/ --include="*.py"
corepack pnpm --dir packages/mcp-server run test:unit
```

PowerShell:
```powershell
Select-String -Path packages/mcp-server/src/**/*.py -Pattern "ipcAvailable|livePcbContext"
corepack pnpm --dir packages/mcp-server run test:unit
```

## Acceptance

- [ ] `ipcAvailable` accurately reflects IPC port reachability at call time
- [ ] Tool error messages and server-info agree on IPC state
- [ ] `livePcbContext` updates when a PCB is opened in KiCad
- [ ] Tests pass
- [ ] Build passes
- [ ] CI is green
- [ ] Evidence added to this issue

### LINKED ISSUE #225: [BUG] pcb_get_layers returns empty result for new KiCad 10.0 project — file-backed parser ignores default layers
## Problem

`pcb_get_layers` returns "No enabled layers were found in the file-backed fallback" when called on a freshly created KiCad 10.0.3 project. KiCad always provisions default copper and non-copper layers (F.Cu, B.Cu, F.SilkS, B.SilkS, Edge.Cuts, etc.) for any new board file, but the file-backed parser cannot extract layer information from a minimal `.kicad_pcb` file.

**Minimal PCB file content:**
```
(kicad_pcb (version 20260206) (generator "pcbnew") (generator_version "10.0")
)
```

**Actual result:**
```
No enabled layers were found in the file-backed fallback.
Diagnostics:
- Source: file-backed
- Fallback status: using file-backed .kicad_pcb parser
```

**Expected result:** At minimum, the default KiCad layer set should be returned (F.Cu, B.Cu, F.SilkS, B.SilkS, F.Mask, B.Mask, Edge.Cuts, etc.), or an explicit message stating "Layer configuration is implicit — showing KiCad defaults."

**Environment:**
- KiCad MCP Pro: v3.5.2
- KiCad CLI: 10.0.3
- OS: Windows 11
- Date: 27/05/2026

## Required changes

1. When the file-backed parser finds no explicit `(layers ...)` section, return the standard KiCad default layer set (2-layer: F.Cu, B.Cu + standard non-copper layers).
2. Document in diagnostics when defaults are being used vs. parsed from file.

## Validation

```bash
grep -rn "get_layers\|parse_layers\|enabled_layers" packages/mcp-server/src/ --include="*.py"
corepack pnpm --dir packages/mcp-server run test:unit
```

PowerShell:
```powershell
Select-String -Path packages/mcp-server/src/**/*.py -Pattern "get_layers|parse_layers|enabled_layers"
corepack pnpm --dir packages/mcp-server run test:unit
```

## Acceptance

- [ ] `pcb_get_layers` returns a non-empty layer list for a minimal new-project `.kicad_pcb`
- [ ] Diagnostics indicate "using KiCad defaults" when no explicit layers section exists
- [ ] Tests cover the minimal-file edge case
- [ ] Tests pass
- [ ] Build passes
- [ ] CI is green

### LINKED ISSUE #227: [FEATURE] Add empty-project detection and onboarding guidance in quality gate tools
## Problem

When an MCP client connects to a freshly created KiCad project, the quality gate tools return accurate but unhelpful results. For example, `project_quality_gate` returns `BLOCKED` with a long list of gates, but doesn't explain that the project is simply empty and needs initial design work.

This is confusing for new users who create a project and immediately run quality checks. The tool output reads like there are serious problems, when in reality the project just hasn't been started yet.

**Current behavior:**
```
Project quality gate: BLOCKED
- Blocking issues remain. Do not treat this design as production-ready yet.
WARN: Design intent not set
PCB quality gate: FAIL - DRC violations: 1 (invalid_outline)
Placement quality gate: BLOCKED - No PCB footprints found
```

**Expected behavior:**
```
Project quality gate: NEW PROJECT
- This appears to be a freshly created project with no components or layout.
- Suggested next steps:
  1. Add symbols to the schematic
  2. Draw a board outline on Edge.Cuts
  3. Set design intent with project_set_design_intent()
  4. Run this gate again after initial design work
```

**Environment:**
- KiCad MCP Pro: v3.5.2
- KiCad CLI: 10.0.3
- OS: Windows 11
- Date: 27/05/2026

## Required changes

1. Detect "empty project" state: 0 symbols, 0 footprints, 0 nets, no board outline.
2. When detected, return a distinct gate status (e.g., `NEW_PROJECT` or `EMPTY`) with onboarding guidance.
3. Include actionable next steps with relevant tool names.
4. Preserve existing gate behavior for non-empty projects.

## Validation

```bash
corepack pnpm --dir packages/mcp-server run test:unit
```

PowerShell:
```powershell
corepack pnpm --dir packages/mcp-server run test:unit
```

## Acceptance

- [ ] Empty projects receive distinct gate status with guidance
- [ ] Non-empty projects are unaffected
- [ ] Next-step suggestions reference correct tool names
- [ ] Tests pass
- [ ] Build passes
- [ ] CI is green

### LINKED ISSUE #229: [TEST] Add IPC state transition tests — verify server-info consistency across IPC lifecycle
## Problem

As of 27/05/2026, there is no test coverage for the IPC state lifecycle. The `kicad_get_server_info` endpoint reports `ipcAvailable`, `livePcbContext`, and `liveSchematicContext`, but these values are not validated against actual IPC state transitions:

1. **Startup without KiCad** → all three should be `false`
2. **KiCad opened, no project loaded** → `ipcAvailable` should be `true`, contexts `false`
3. **KiCad with PCB open** → `ipcAvailable: true`, `livePcbContext: true`
4. **KiCad with both PCB+SCH open** → all three `true`
5. **KiCad closed** → all three should return to `false`

Currently observed: `ipcAvailable: false` even when tool-level diagnostics say "IPC is reachable" — suggesting the server-info probe is stale or cached from startup.

## Required changes

1. Create integration tests (or document as manual test plan) covering IPC state transitions.
2. Test that `kicad_get_server_info` values are consistent with tool-level IPC diagnostics.
3. Test that opening/closing KiCad documents updates the context flags.
4. If automated IPC tests are not practical (requires running KiCad), create a mock-based unit test that validates the probe logic and a documented manual test plan.

## Validation

```bash
grep -rn "ipc\|IPC\|livePcbContext" packages/mcp-server/tests/ --include="*.py"
corepack pnpm --dir packages/mcp-server run test:unit
```

PowerShell:
```powershell
Select-String -Path packages/mcp-server/tests/**/*.py -Pattern "ipc|IPC|livePcbContext"
corepack pnpm --dir packages/mcp-server run test:unit
```

## Acceptance

- [ ] IPC state transition logic covered by unit test (with mocks) or documented manual test plan
- [ ] `kicad_get_server_info` consistency with tool diagnostics validated
- [ ] Test or manual plan covers all 5 transition states
- [ ] Tests pass
- [ ] Build passes
- [ ] CI is green

### PR DESCRIPTION
**Pull request by oaslananka**
Resolves multiple MCP core fallback behaviors, quality gate onboarding guides, and lifecycle transition checks.

Closes #223
Closes #225
Closes #227
Closes #229
Closes #252

### CODE REVIEW

File: apps/vscode-extension/src/commands/qualityGateCommands.ts [hunk 1]
```diff
@@ -15,15 +15,29 @@ export function registerQualityGateCommands(
...
   ];
 }
+
+type QualityGateCommandArg =
+  | QualityGateResult
+  | { kind: 'gate'; gate: QualityGateResult };
+
+function resolveGateArg(gate: QualityGateCommandArg): QualityGateResult {
+  return 'kind' in gate && gate.kind === 'gate' ? gate.gate : (gate as QualityGateResult);
+}
```
[the remark this answers is not in the document]
Comment (oaslananka) [reply apps/vscode-extension/src/commands/qualityGateCommands.ts:43]: Done! Integrated the safe nullish check for `gate`.

File: apps/vscode-extension/src/ai/chatHtml.ts [hunk 2]
```diff
@@ -380,7 +404,9 @@ export function buildChatHtml(options: ChatHtmlOptions): string {
       tokenEstimate: document.getElementById('token-estimate'),
       send: document.getElementById('send'),
       cancel: document.getElementById('cancel'),
-      toggleContext: document.getElementById('toggle-context')
+      toggleContext: document.getElementById('toggle-context'),
+      quotaBanner: document.getElementById('quota-banner'),
+      quotaBannerText: document.getElementById('quota-banner-text')
```
[the remark this answers is not in the document]
Comment (oaslananka) [reply apps/vscode-extension/src/ai/chatHtml.ts:408]: Done! Removed the duplicate `quotaBanner` property.

File: apps/vscode-extension/src/mcp/mcpToolsProvider.ts [hunk 3]
```diff
@@ -648,9 +704,35 @@ function modeIcon(mode: string): string {
...
+  if (!serverInfo) {
+    return false;
+  }
+  return (
+    !serverInfo.kicad.ipcAvailable ||
+    !serverInfo.kicad.livePcbContext ||
+    !serverInfo.kicad.liveSchematicContext ||
+    !serverInfo.capabilities.livePcbRead ||
+    !serverInfo.capabilities.liveSchematicRead
+  );
```
[the remark this answers is not in the document]
Comment (oaslananka) [reply apps/vscode-extension/src/mcp/mcpToolsProvider.ts:735]: Done! Added optional chaining on `serverInfo` checks.
